http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +544,124 @@ private void addScopes(String scopes, Set<String> allScopes) {
             allScopes.addAll(Arrays.asList(scopes.split("\\s+")));
         }
     }
+
+    private String convertToRequestParameter(URIBuilder redirectUriBuilder, String redirectUri, String state, List<NameValuePair> forwardedQueryParams) throws JoseException, IOException {
+        redirectUriBuilder.addParameter(SCOPE, OIDC_SCOPE);
+
+        JwtClaims jwtClaims = new JwtClaims();
+        jwtClaims.setIssuer(deployment.getIssuerUrl());
+        jwtClaims.setAudience(deployment.getProviderUrl());
+
+        for ( NameValuePair parameter: forwardedQueryParams) {
+            jwtClaims.setClaim(parameter.getName(), parameter.getValue());
+        }
+        jwtClaims.setClaim(STATE, state);
+        jwtClaims.setClaim(REDIRECT_URI, redirectUri);
+        jwtClaims.setClaim(RESPONSE_TYPE, CODE);
+        jwtClaims.setClaim(CLIENT_ID, deployment.getResourceName());
+
+        // sign JWT first before encrypting
+        JsonWebSignature signedRequest = signRequest(jwtClaims);
+
+        // Encrypting optional
+        if (deployment.getRequestObjectEncryptionAlgorithm() != null && !deployment.getRequestObjectEncryptionAlgorithm().isEmpty() &&
```

Comment:
Can it be the empty string? Is that a valid value?

Reply:
Yes, the default values for enc and alg are undefined. So, if the user does not choose to encrypt the JWT, which is optional, then these values would be null.

Reply:
All comments with a thumbs up till here have been addressed in https://github.com/wildfly-security/wildfly-elytron/pull/1984/commits/a9415cf529b74bfdca575f43c53a2d5a5eb8d3f2
For the ones without a followup questions are pending. 
Thanks @fjuma